### ISSUE TITLE
Manage fails to install on SNO due to ReadWriteMany PVC

### ISSUE THREAD

**Issue by belfortmaycon** [ibm-mas/installer#369]
Hi team, I'm installing MAS 8.10 in an SNO cluster, and when running the pipeline in the app-cfg-manage task, it fails due to the PVCs for doclinks, bim, and jms being created as RWM and not as RWO.

I can see in the pipeline YAML that the variables are RWO.

```yaml
 - name: mas_app_settings_demodata
  value: 'false'
- name: mas_appws_bindings_jdbc_manage
  value: workspace-application
- name: mas_app_settings_persistent_volumes_flag
  value: 'true'
- name: mas_app_settings_jms_queue_pvc_storage_class
  value: odf-lvm-vg1
- name: mas_app_settings_jms_queue_pvc_accessmode
  value: ReadWriteOnce
- name: mas_app_settings_bim_pvc_storage_class
  value: odf-lvm-vg1
- name: mas_app_settings_bim_pvc_accessmode
  value: ReadWriteOnce
- name: mas_app_settings_doclinks_pvc_storage_class
  value: odf-lvm-vg1
- name: mas_app_settings_doclinks_pvc_accessmode
  value: ReadWriteOnce
- name: mas_app_settings_server_bundles_size
  value: snojms
```

The PVC is in Pending status, and the YAML is ReadWriteMany
```yaml
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  resourceVersion: '6459224'
  name: tst-test-manage-doclinks
  uid: 957b8000-87b0-4add-8c57-9d85cc5eccdb
  creationTimestamp: '2023-07-17T08:06:52Z'
  namespace: mas-tst-manage
  ownerReferences:
    - apiVersion: apps.mas.ibm.com/v1
      kind: ManageDeployment
      name: manage-maxinst
      uid: 3aff2402-9603-4c87-9005-0e1ecc2b4081
  finalizers:
    - kubernetes.io/pvc-protection
  labels:
    app.kubernetes.io/instance: tst
    app.kubernetes.io/managed-by: ibm-mas-manage
    app.kubernetes.io/name: ibm-mas-manage
spec:
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 20Gi
  storageClassName: odf-lvm-vg1
  volumeMode: Filesystem
status:
  phase: Pending
```

The WorkSpace is also using RWO.
```yaml
apiVersion: apps.mas.ibm.com/v1
kind: ManageWorkspace
metadata:
  resourceVersion: '6472508'
  name: tst-test
  uid: 1f9a0577-ea80-4a2f-a98f-3a1a7422e762
  creationTimestamp: '2023-07-17T08:03:26Z'
  generation: 1
  managedFields:
    - apiVersion: apps.mas.ibm.com/v1
      [...]
      manager: OpenAPI-Generator
      operation: Update
      time: '2023-07-17T08:03:26Z'
    - apiVersion: apps.mas.ibm.com/v1
      manager: ansible-operator
      operation: Update
      time: '2023-07-17T08:03:26Z'
    - apiVersion: apps.mas.ibm.com/v1
  namespace: mas-tst-manage
  finalizers:
    - manageworkspace.apps.mas.ibm.com/finalizer
  labels:
    mas.ibm.com/applicationId: manage
    mas.ibm.com/instanceId: tst
    mas.ibm.com/workspaceId: test
spec:
  bindings:
    jdbc: workspace-application
  components:
    base:
      version: latest
    health:
      version: latest
  settings:
    aio:
      install: false
    db:
      dbSchema: maximo
      maxinst:
        bypassUpgradeVersionCheck: false
        db2Vargraphic: true
        demodata: false
        indexSpace: MAXINDEX
        tableSpace: MAXDATA
    deployment:
      buildTag: latest
      mode: up
      persistentVolumes:
        - accessModes:
            - ReadWriteOnce
          mountPath: /DOCLINKS
          pvcName: manage-doclinks
          size: 20Gi
          storageClassName: odf-lvm-vg1
        - accessModes:
            - ReadWriteOnce
          mountPath: /bim
          pvcName: manage-bim
          size: 20Gi
          storageClassName: odf-lvm-vg1
        - accessModes:
            - ReadWriteOnce
          mountPath: /jms
          pvcName: manage-jms
          size: 20Gi
          storageClassName: odf-lvm-vg1
```

The SNO will accept only RWO, is there something I can do to fix it?

Thank you.

**Comment by whitfiea** [comment]
@Aymen-Ben-S @sbudhira could you comment on this SNO error please?

**Comment by Aymen-Ben-S** [comment]
I'm adding @andrercm as I remember he worked on configuring access modes.

**Comment by andrercm** [comment]
@belfortmaycon would you please inform the MAS cli version you are using? 

So from what I can tell from the information you provided the ManageWorkspace CR is being properly configured, which means the cli is passing the correct and expected parameters to Manage installation:

```
      persistentVolumes:
        - accessModes:
            - ReadWriteOnce
          mountPath: /DOCLINKS
          pvcName: manage-doclinks
          size: 20Gi
          storageClassName: odf-lvm-vg1
        - accessModes:
            - ReadWriteOnce
          mountPath: /bim
          pvcName: manage-bim
          size: 20Gi
          storageClassName: odf-lvm-vg1
        - accessModes:
            - ReadWriteOnce
          mountPath: /jms
          pvcName: manage-jms
          size: 20Gi
          storageClassName: odf-lvm-vg1
```

Can you tell if just the doclinks PVC has been wrongly created or if the same happened to the other 2 PVCs? (bim and jms as well)

**Comment by belfortmaycon** [comment]
Hi @andrercm, I'm not sure, but when I got this error I think it was 6.0.2 or 6.1. I have tested it in a MNO cluster and it worked fine.

The same error happened for all 3 PVCs, bim, jms and doclinks.

I'm gonna install another environment later this week or early next week, and I can try with the latest mascli.

**Comment by andrercm** [comment]
@belfortmaycon sure thing, let me know if you still face the same problem and we can debug.

**Comment by belfortmaycon** [comment]
Hi @andrercm,

After updating to the latest cli version, it worked as expected.

Thank you.

**Comment by belfortmaycon** [comment]
Hi @andrercm,

I have another environment where I have installed MAS using the CLI, and I'm still getting this error if the pod [INSTANCE]-entitymgr-ws is restarted.

**Comment by andrercm** [comment]
@belfortmaycon please provide the following command outputs:

```
export MAS_INSTANCE_ID=replace-with-instance-id
oc get pvc -n mas-$MAS_INSTANCE_ID-manage
```

and 

```
export MAS_INSTANCE_ID=<replace-with-workspace-id>
export MAS_WORKSPACE_ID=<replace-with-instance-id>
oc get ManageWorkspace $MAS_INSTANCE_ID-$MAS_WORKSPACE_ID -n mas-$MAS_INSTANCE_ID-manage -o yaml
```

**Comment by andrercm** [comment]
And just to understand the overall scenario, what's the reason or condition that is causing the pod entitymgr-ws to be restarted? If this pod is not restarted, then access mode for all three PVC remains on `ReadWriteOnce ` as expected ?

**Comment by andrercm** [comment]
Closing as no feedback for the past two weeks.. @belfortmaycon let us know if the issue needs to be reopened.

**Comment by belfortmaycon** [comment]
Hi Andre,

I’m on leave without access to my environment. Could you please leave this
issue open? I’ll be back on the 2nd October.

Thank you.

**Comment by belfortmaycon** [comment]
Hi @andrercm,

Below are the output commands you asked for.

> @belfortmaycon please provide the following command outputs:
> 
> ```
> export MAS_INSTANCE_ID=replace-with-instance-id
> oc get pvc -n mas-$MAS_INSTANCE_ID-manage
> ```
Output:
```
❯ export MAS_INSTANCE_ID=tst
❯ oc get pvc -n mas-$MAS_INSTANCE_ID-manage
NAME                       STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
tst-test-manage-bim        Bound    pvc-e031f137-66e6-43ed-ae88-39f241e54f7f   20Gi       RWO            odf-lvm-vg1    76d
tst-test-manage-doclinks   Bound    pvc-ffa441e9-634b-4ca7-8033-80b973bc8c70   20Gi       RWO            odf-lvm-vg1    76d
tst-test-manage-jms        Bound    pvc-7be29f8c-b1dc-48ae-9dfd-f91a6420b09c   20Gi       RWO            odf-lvm-vg1    76d
```
> 
> and
> 
> ```
> export MAS_INSTANCE_ID=<replace-with-workspace-id>
> export MAS_WORKSPACE_ID=<replace-with-instance-id>
> oc get ManageWorkspace $MAS_INSTANCE_ID-$MAS_WORKSPACE_ID -n mas-$MAS_INSTANCE_ID-manage -o yaml
> ```
Output:
```
❯ export MAS_WORKSPACE_ID=test
❯ oc get ManageWorkspace $MAS_INSTANCE_ID-$MAS_WORKSPACE_ID -n mas-$MAS_INSTANCE_ID-manage -o yaml
apiVersion: apps.mas.ibm.com/v1
kind: ManageWorkspace
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: '{"apiVersion":"apps.mas.ibm.com/v1","kind":"ManageWorkspace","metadata":{"labels":{"mas.ibm.com/applicationId":"manage","mas.ibm.com/instanceId":"tst","mas.ibm.com/workspaceId":"test"},"name":"tst-test","namespace":"mas-tst-manage"},"spec":{"bindings":{"jdbc":"workspace-application","optimizer":"workspace"},"components":{"base":{"version":"latest"},"health":{"version":"latest"},"hse":{"version":"latest"}},"settings":{"aio":{"install":false},"db":{"dbSchema":"maximo","maxinst":{"bypassUpgradeVersionCheck":false,"db2Vargraphic":true,"demodata":false,"indexSpace":"MAXINDEX","tableSpace":"MAXDATA"},"upgrade":{"oldAuthType":"local"}},"deployment":{"autoGenerateEncryptionKeys":true,"persistentVolumes":[{"accessModes":["ReadWriteOnce"],"mountPath":"/DOCLINKS","pvcName":"manage-doclinks","size":"20Gi","storageClassName":"odf-lvm-vg1"},{"accessModes":["ReadWriteOnce"],"mountPath":"/bim","pvcName":"manage-bim","size":"20Gi","storageClassName":"odf-lvm-vg1"},{"accessModes":["ReadWriteOnce"],"mountPath":"/jms","pvcName":"manage-jms","size":"20Gi","storageClassName":"odf-lvm-vg1"}],"serverBundles":[{"additionalServerConfig":{"secretName":"test-manage-d--sb0--asc--sn"},"bundleType":"all","isDefault":true,"isMobileTarget":true,"isUserSyncTarget":true,"name":"all","replica":1,"routeSubDomain":"all"},{"additionalServerConfig":{"secretName":"test-manage-d--sb1--asc--sn"},"bundleType":"standalonejms","isDefault":false,"isMobileTarget":false,"isUserSyncTarget":false,"name":"jms","replica":1,"routeSubDomain":"jms"}]}}}}'
  creationTimestamp: "2023-07-17T08:03:26Z"
  finalizers:
  - manageworkspace.apps.mas.ibm.com/finalizer
  generation: 16
  labels:
    mas.ibm.com/applicationId: manage
    mas.ibm.com/instanceId: tst
    mas.ibm.com/workspaceId: test
  name: tst-test
  namespace: mas-tst-manage
  resourceVersion: "86751762"
  uid: 1f9a0577-ea80-4a2f-a98f-3a1a7422e762
spec:
  bindings:
    jdbc: workspace-application
  components:
    base:
      version: latest
    health:
      version: latest
    hse:
      version: latest
  settings:
    aio:
      install: false
    db:
      dbSchema: maximo
      maxinst:
        bypassUpgradeVersionCheck: false
        db2Vargraphic: true
        demodata: false
        indexSpace: MAXINDEX
        tableSpace: MAXDATA
      upgrade:
        oldAuthType: local
    deployment:
      autoGenerateEncryptionKeys: true
      buildTag: latest
      mode: up
      persistentVolumes:
      - accessModes:
        - ReadWriteOnce
        mountPath: /DOCLINKS
        pvcName: manage-doclinks
        size: 20Gi
        storageClassName: odf-lvm-vg1
      - accessModes:
        - ReadWriteOnce
        mountPath: /bim
        pvcName: manage-bim
        size: 20Gi
        storageClassName: odf-lvm-vg1
      - accessModes:
        - ReadWriteOnce
        mountPath: /jms
        pvcName: manage-jms
        size: 20Gi
        storageClassName: odf-lvm-vg1
      serverBundles:
      - additionalServerConfig:
          secretName: test-manage-d--sb0--asc--sn
        bundleType: all
        isDefault: true
        isMobileTarget: true
        isUserSyncTarget: true
        name: all
        replica: 1
        routeSubDomain: all
      - additionalServerConfig:
          secretName: test-manage-d--sb1--asc--sn
        bundleType: standalonejms
        isDefault: false
        isMobileTarget: false
        isUserSyncTarget: false
        name: jms
        replica: 1
        routeSubDomain: jms
      serverTimezone: GMT
    languages:
      baseLang: EN
      secondaryLangs: []
status:
  bindings:
    jdbc: workspace-application
  components:
    acm:
      enabled: false
      version: 8.1.4
    anywhere:
      enabled: false
      version: 8.0.2
    aviation:
      enabled: false
      version: 8.1.5
    base:
      enabled: true
      version: 8.6.2
    civil:
      enabled: false
      version: 8.5.2
    envizi:
      enabled: false
      version: 8.0.0
    health:
      enabled: true
      url: /maximo/oslc/graphite/relengineer
      version: 8.8.2
    hse:
      enabled: true
      version: 8.1.5
    nuclear:
      enabled: false
      version: 8.1.3
    oilandgas:
      enabled: false
      version: 8.1.4
    oracleadapter:
      enabled: false
      version: 8.2.0
    sapadapter:
      enabled: false
      version: 8.2.0
    serviceprovider:
      enabled: false
      version: 8.1.5
    spatial:
      enabled: false
      version: 8.6.2
    transportation:
      enabled: false
      version: 8.1.4
    tririga:
      enabled: false
      version: 8.0.0
    utilities:
      enabled: false
      version: 8.1.4
    workday:
      enabled: false
      version: 8.1.0
  conditions:
  - lastTransitionTime: "2023-09-29T21:56:46Z"
    message: Manage deployed in workspace test
    reason: Ready
    status: "True"
    type: Ready
  - lastTransitionTime: "2023-09-29T21:56:46Z"
    message: Build generated and tagged with 20230719T034239-8.6
    reason: Build completed
    status: "True"
    type: BuildReady
  - lastTransitionTime: "2023-09-29T21:56:46Z"
    message: ' UpdateDB/Maxinst: Completed. Deployment all: 1 of 1 replicas ready,
      Deployment jms: 1 of 1 replicas ready,'
    reason: Ready
    status: "True"
    type: DeploymentReady
  - lastTransitionTime: "2023-09-29T21:56:46Z"
    message: Awaiting next reconciliation
    reason: Successful
    status: "True"
    type: DeploymentCR
  - lastTransitionTime: "2023-08-30T19:49:30Z"
    message: ""
    reason: ""
    status: "False"
    type: Failure
  - ansibleResult:
      changed: 3
      completion: 2023-10-02T00:07:15.380896
      failures: 0
      ok: 204
      skipped: 99
    lastTransitionTime: "2023-08-30T19:49:51Z"
    message: Awaiting next reconciliation
    reason: Successful
    status: "True"
    type: Running
  - lastTransitionTime: "2023-10-02T00:07:16Z"
    message: Last reconciliation succeeded
    reason: Successful
    status: "True"
    type: Successful
  milestones:
    activated:
      anonymousId: b04b1caf-e61a-4f30-8b74-d03e86e733d3
      timestamp: "2023-07-17T08:04:19.661182+00:00"
      transactionId: 81a77b82-363a-4189-9c2a-2ca2a6671951
  settings:
    aio:
      install: false
    customization: {}
    customizationList: []
    db:
      dbSchema: maximo
      maxinst:
        bypassUpgradeVersionCheck: false
        db2Vargraphic: true
        demodata: false
        indexSpace: MAXINDEX
        tableSpace: MAXDATA
      upgrade:
        oldAuthType: local
    deployment:
      buildTag: latest
      ephemeralStorage:
        limits:
          adminBuild: 100Gi
          serverBundleBuild: 100Gi
        requests:
          adminBuild: 30Gi
          serverBundleBuild: 30Gi
      importedCerts: []
      mode: up
      persistentVolumes:
      - accessModes:
        - ReadWriteOnce
        mountPath: /DOCLINKS
        pvcName: manage-doclinks
        size: 20Gi
        storageClassName: odf-lvm-vg1
      - accessModes:
        - ReadWriteOnce
        mountPath: /bim
        pvcName: manage-bim
        size: 20Gi
        storageClassName: odf-lvm-vg1
      - accessModes:
        - ReadWriteOnce
        mountPath: /jms
        pvcName: manage-jms
        size: 20Gi
        storageClassName: odf-lvm-vg1
      serverBundles:
      - additionalServerConfig:
          secretName: test-manage-d--sb0--asc--sn
        bundleType: all
        isDefault: true
        isMobileTarget: true
        isUserSyncTarget: true
        name: all
        replica: 1
        routeSubDomain: all
      - additionalServerConfig:
          secretName: test-manage-d--sb1--asc--sn
        bundleType: standalonejms
        isDefault: false
        isMobileTarget: false
        isUserSyncTarget: false
        name: jms
        replica: 1
        routeSubDomain: jms
      serverTimezone: GMT
      useMasSmtpcfg: true
    languages:
      baseLang: EN
      secondaryLangs: []
    manualCertMgmt: false
    resources:
      manageAdmin:
        limits:
          cpu: "2"
          memory: 4Gi
        requests:
          cpu: "0.5"
          memory: 500Mi
      serverBundles:
        limits:
          cpu: "6"
          memory: 10Gi
        requests:
          cpu: "0.5"
          memory: 2Gi
  version:
    reconciled: ""
  versions:
    reconciled: 8.6.2
```

**Comment by belfortmaycon** [comment]
> And just to understand the overall scenario, what's the reason or condition that is causing the pod entitymgr-ws to be restarted? If this pod is not restarted, then access mode for all three PVC remains on `ReadWriteOnce ` as expected ?

Last time, there was a failure in the host machine, so we had to restart the VM. During the installation, because of the mismatch with the access modes, we manually changed the file `roles/manage-deployment/templates/maxinst/deployment/persistent-claim.j2.yml` in line 18 from `ReadWriteMany` to `ReadWriteOnce` as it seems the if condition wasn't working. But in a scenario like the machine restarted, it returns to `ReadWriteMany`, and the ManageWorkspace instance fails.
